### PR TITLE
Parametrize `XYZAssembly` plotting tests 

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Unify and parametrize prop3d positioning tests for subclasses of  `XYZAssembly`  and add new `show_labels` tests.

This is a follow-up to #6394 and #6379.

### CODE REVIEW

**Comment by user27182** [comment]
The parameterization renames the files _and_ there are some tweaks made to the tests (e.g. use of `add_axes_at_origin` and white text for all labels). This makes the diff a bit hard to follow. If desired I can create an intermediate PR to modify the tests first so that parameterization only renames the files and shows a better diff.

**Comment by tkoyama010** [comment]
> The parameterization renames the files _and_ there are some tweaks made to the tests (e.g. use of `add_axes_at_origin` and white text for all labels). This makes the diff a bit hard to follow. If desired I can create an intermediate PR to modify the tests first so that parameterization only renames the files and shows a better diff.

Thanks. It is easy enough to understand.
Thanks again for all the Pull Requests.

**Review by tkoyama010 — APPROVED** [review]
LGTM
